examples/image_ocr.py

```diff
@@ -286,20 +303,18 @@ def next_val(self):
                 self.cur_val_index = self.val_split + self.cur_val_index % 32
             yield ret
 
-    def on_train_begin(self, logs={}):
-        # translational invariance seems to be the hardest thing
-        # for the RNN to learn, so start with <= 4 letter words.
-        self.build_word_list(16000, 4, 1)
-
     def on_epoch_begin(self, epoch, logs={}):
-        # After 10 epochs, translational invariance should be learned
-        # so start feeding longer words and eventually multiple words with spaces
-        if epoch == 10:
-            self.build_word_list(32000, 8, 1)
-        if epoch == 20:
-            self.build_word_list(32000, 8, 0.6)
-        if epoch == 30:
-            self.build_word_list(64000, 12, 0.5)
+        cur_epoch = epoch + self.start_epoch
+        # rebind the paint function to implement curriculum learning
+        if cur_epoch == 3:
+            self.paint_func = lambda text: paint_text(text, self.img_w, self.img_h,
+                                                      rotate=False, ud=True, multi_fonts=False)
+        if cur_epoch == 6:
+            self.paint_func = lambda text: paint_text(text, self.img_w, self.img_h,
+                                                      rotate=False, ud=True, multi_fonts=True)
+        if cur_epoch == 12:
```

Comment:
Ok, I see..i hard-coded "20" in the TextGenerator constructor so as submitted it works fine, but if you start changing the start_epoch it doesn't like it.  Like I said, this was a hacky addition. I think using "initial_epoch" I can make it cleaner.